solutions/gold/cses-1095.mdx

```diff
@@ -56,4 +56,31 @@ int main() {
 }
 ```
 </CPPSection>
+
+<PySection>
+```py
+import sys
+
+input = sys.stdin.readline
+takeInt = lambda: int(input())
+takeInts = lambda: map(int, input().split())
+takeList = lambda: list(map(int, input().split()))
+mod = int(1e9) + 7
+
+
+def power(base, exponent, mod):
```

Comment:
python has builtin modular exponentiation, use that please

Reply:
yeah but i want to show the process of binary exponentiation to perform in python so i did it. do i need to change it now

Reply:
yeah

Reply:
the whole point of the problem is to use modular exponentiation. I think he should keep it.

Reply:
look here: https://usaco.guide/gold/modular?lang=py#solution---exponentiation

if people wanna look at the method behind it they can look up how the `pow` works